fix: github action missing Trivy scan on sidecar image

zdtsw opened the pull request:
- Trivy scan should be done on both images
- move "make build" before "make test" to have quick exit
- use "make" target to install k-v cache dependency
- use Github Action than download lychee binary

elevran commented:
@zdtsw DCO is missing, Please see https://github.com/llm-d/llm-d-inference-scheduler/pull/481/checks?check_run_id=56199228166

zdtsw commented:
> > if we need more confirmation regarding python header file, should I split this PR into two so get the "missing Trivy scan" out first?
> 
> No further confirmation is needed and not need to split PR. I think it is safest to go back to what you originally had and remove Python from Makefile's install-dependencies target. We have #470 to track Makefile issue on dependencies. /unhold

Thanks,  I've reverted the changes for python3-dev and updated the commit message
